### PR TITLE
refactor: Client Credential Authentication with Application ID and Secret

### PR DESCRIPTION
**Pull request by yohei-comvex**
## Description

User key based authentication agains Orchestrator API has been deprecated.

Refactored to Application ID/Secret based authentication.

## Confirmation

Don't we need fallback to the old authentication logic if it fails since `uipath-go` Client is not publc for now so may be we don't care of backward compatibility?

## References
https://comvex.myjetbrains.com/youtrack/agiles/107-69/108-632?issue=DGM2-10147

### CODE REVIEW

File: examples/example.go [hunk 1]
```diff
@@ -56,7 +73,7 @@ func (e *Examples) GetAssetById() (uipath.Asset, error) {
 
 	aHandler := uipath.AssetHandler{
 		Client:   e.Client,
-		FolderId: uint(folderID),
+		FolderId: folderID,
```
Comment (hiromi-sannomiya) [inline examples/example.go:76]: The folderID appears to be an int, does that mean I don't have to cast it to uint?

**Comment by cliff-comvex** [comment]
> Don't we need fallback to the old authentication logic if it fails since uipath-go Client is not publc for now so may be we don't care of backward compatibility?

@yohei-comvex yeah I think we don't need backward compatibility. So I am also thinking maybe we should bump this version to v1?

**Review by hiromi-sannomiya — APPROVED** [review]
I have one question and it appears to be fine.

**Comment by yohei-comvex** [comment]
>So I am also thinking maybe we should bump this version to v1?

@cliff-comvex ~do you mean to move all files under `v1` directory?~

> yeah I think we don't need backward compatibility

so, do you think we can remove Deprecated methods and env variables like `UIPATH_USER_KEY` from the codebase and vault?

am bit wondering if there's any possibility to fail fetching auth token(s) while production deployments or application token will be revoked/updated by human errors or security reason(s).

**Review by Stefancomvex — APPROVED** [review]

File: examples/example.go (hunk 1 above)
Comment (yohei-comvex) [reply examples/example.go:76]: you can cast on the variable initalization

**Comment by yohei-comvex** [comment]
@cliff-comvex 

>  So I am also thinking maybe we should bump this version to v1?

maybe we will do this in another PR since this pr is blocking other tasks.

**Comment by anderson-comvex** [comment]
@yohei-comvex Release this version as version 1.0.0 in the package.yml

**Comment by anderson-comvex** [comment]
Add a changelog and package.yml

**Review by anderson-comvex — APPROVED** [review]

**Comment by yohei-comvex** [comment]
> yeah I think we don't need backward compatibility

Added an issue #9 so we can cope with it later on

**Review by cliff-comvex — APPROVED** [review]

**Pull request merged**